tds-dll-mysql/src/main/java/tds/dll/mysql/ItemSelectionDLL.java

```diff
@@ -154,92 +163,101 @@ public SingleDataResultSet AA_GetNextItemCandidates_SP(
 				result.addRecords(resultList);
 				return result;
 			}
-			// found the segment to work on
-			// go to next
-			else if (!_AA_IsSegmentSatisfied_FN(connection, oppkey, segment)) {
-				// TODO: check, what value returns this function, after
-				// isSegmentSatisfied (oppkey, segment) will be implemented
-				break;
-			} else {// segment is satisfied, mark and continue loop, NOTE:
+
+			while(currentRecord != null && currentRecord.hasNext()) {
+				record = currentRecord.next();
+
+				if (record != null) {
+					segment = record.<Integer> get("SegmentPosition");
+					segmentKey = record.<String> get("_efk_Segment");
+					segmentID = record.<String> get("segmentID");
+					algorithm = record.<String> get("algorithm");
+					ftcnt = record.<Integer> get("ftItemCnt");
+				}
+				// found the segment to work on
```

Comment:
Should this `continue` or a `break` if a segment isn't found?  Asking based on the comment 'found the segment to work on'